src/main/java/MainWindow.java

```diff
@@ -0,0 +1,76 @@
+import java.util.concurrent.CompletableFuture;
+
+import javafx.application.Platform;
+import javafx.fxml.FXML;
+import javafx.scene.control.Button;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.image.Image;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.VBox;
+
+
+/**
+ * Controller for MainWindow. Provides the layout for the other controls.
+ */
+public class MainWindow extends AnchorPane {
+    @FXML
+    private ScrollPane scrollPane;
+    @FXML
+    private VBox dialogContainer;
+    @FXML
+    private TextField userInput;
+    @FXML
+    private Button sendButton;
+
+    private Duke duke;
+
+    private final Image userImage = new Image(this.getClass().getResourceAsStream("/images/whip.png"));
+    private final Image dukeImage = new Image(this.getClass().getResourceAsStream("/images/cryLaugh.png"));
+
+    @FXML
+    public void initialize() {
+        scrollPane.vvalueProperty().bind(dialogContainer.heightProperty());
+    }
+
+    /**
+     * Initialises Duke and displays the opening message from Duke.
+     *
+     * @param d Duke object.
+     */
+    public void setDuke(Duke d) {
+        duke = d;
+        dialogContainer.getChildren().add(
+                DialogBox.getDukeDialog(duke.getOpening(), dukeImage)
+        );
+    }
+
+    /**
+     * Creates two dialog boxes by default, one echoing user input and the other containing Duke's reply
+     * and then appends them to the dialog container. Clears the user input after processing.
```

Comment:
Could add why there is a use of CompletableFuture. I actually would like to know also haha